File: Chapter11/HelpDeskCS/HelpDeskCS/HelpDeskCS.Server/DataSources/ApplicationData/IssueDocument.lsml.cs
using Microsoft.LightSwitch;
using System.Text;
using System.Linq;
using System.Collections.Generic;
using System;

namespace LightSwitchApplication
{
    public partial class IssueDocument
    {
        //Listing 11-10. Validating file sizes

        partial void IssueFile_Validate(EntityValidationResultsBuilder results)
        {
            if (IssueFile != null)
            {
                var sizeInMB = IssueFile.Length / 1048576;
                if (sizeInMB > 2)
                {
                    results.AddPropertyError("File cannot be > 2MB");
                }
            }
        }
    }
}

File: Chapter11/HelpDeskVB/HelpDeskVB/HelpDeskVB.Server/DataSources/ApplicationData/IssueDocument.lsml.vb
Namespace LightSwitchApplication
    Public Class IssueDocument

        'Listing 11-10. Validating file sizes

        Private Sub IssueFile_Validate(results As EntityValidationResultsBuilder)
            If Me.IssueFile IsNot Nothing Then
                Dim sizeInMB = Me.IssueFile.Length / 1048576
                If sizeInMB > 2 Then
                    results.AddPropertyError("File cannot be > 2MB")
                End If
            End If
        End Sub

    End Class
End Namespace
